src/App.tsx

```diff
@@ -1,24 +1,368 @@
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(Errors.NoErrors);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+
+  const delayErrorDisappear = useCallback(() => {
+    setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+  }, []);
+
+  useEffect(() => {
+    getAllTodos(USER_ID)
+      .then(setTodos);
+  }, []);
+
+  const handleFilterSelection = useCallback((value: string) => {
+    setSelectedFilter(value);
+  }, [selectedFilter]);
+
+  const handleChangeInputValue = useCallback((value: string) => {
+    setHasError(false);
+    setTodoTitle(value);
+  }, [todoTitle]);
+
+  const handleAddTodo = async (event: FormEvent<HTMLFormElement>) => {
+    setIsLoading(true);
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setHasError(true);
+      setErrorMessage(Errors.TitleError);
+      delayErrorDisappear();
+      setIsLoading(false);
+
+      return;
+    }
+
+    try {
+      setTempTodo({
+        id: 0,
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      });
+
+      const todoToAdd: Todo = await client.post(`/todos?userId=${USER_ID}`, {
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      });
+
+      setIsLoading(false);
+      setTempTodo(null);
+      setTodos(prevTodos => [...prevTodos, todoToAdd]);
+    } catch {
+      setHasError(true);
+      setErrorMessage(Errors.AddError);
+      delayErrorDisappear();
+    } finally {
+      setTodoTitle('');
+    }
+  };
+
+  const handleTodoDelete = async (todoId: number) => {
+    setDeletingTodoId(todoId);
+    try {
+      await client.delete(`/todos/${todoId}`);
+      setTodos(prevTodos => prevTodos.filter(({ id }) => id !== todoId));
+    } catch {
+      setHasError(true);
+      setErrorMessage(Errors.DeleteError);
+      delayErrorDisappear();
+    } finally {
+      setDeletingTodoId(null);
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const allTodosFromServer = await getAllTodos(USER_ID)
+      .then(completedTodos => completedTodos);
+
+    const activeTodos = allTodosFromServer
+      .filter(({ completed }) => !completed);
+
+    const completedTodos = allTodosFromServer
+      .filter(({ completed }) => completed);
+
+    Promise.all(completedTodos.map(({ id }) => client.delete(`/todos/${id}`)));
+
+    setTodos(activeTodos);
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const visibleTodos = todos.filter(({ completed }) => {
+    switch (selectedFilter) {
+      case Filters.All:
+        return true;
+
+      case Filters.Active:
+        return !completed;
+
+      case Filters.Completed:
+        return completed;
+
+      default:
```

Comment:
Change `return 0;` to `return false;` to match the expected return type of a boolean in the filter function.